src/wallet/wallet2.cpp

```diff
@@ -577,6 +577,13 @@ tools::wallet2::tx_construction_data get_construction_data_with_decrypted_short_
   return construction_data;
 }
 
+uint32_t get_subaddress_clamped_sum(uint32_t idx, uint32_t extra)
+{
+  if (idx > 0xffffffff - extra)
```

Comment:
That looks harder to read and 3 or 4 times as long though.

Reply:
How about `(uint32_t)-1`? It's commonly used in the codebase.

Reply:
I think that's less readable. I'd typically use that as a sentinel value though, where it's clearer.

Reply:
Isn't `0xffffffff` less readable (or foolproof) because it looks pretty similar to `0xfffffff` or `0xfffffffff`?

Reply:
>static constexpr uint32_t uint32_max = std::numeric_limits<uint32_t>::max();
>How about (uint32_t)-1? It's commonly used in the codebase.
>0xfffffff oops, missed a nibble

Yes, I would prefer this because it's explicit and not error-prone: `static constexpr std::uint32_t max = std::numeric_limits<std::uint32_t>::max();`. If this is used frequently elsewhere, how about a singleton definition?